apps/mull-api/src/app/user/user.resolver.ts

```diff
@@ -96,40 +97,51 @@ export class UserResolver {
     return this.userService.deleteUser(id);
   }
 
+  @UseGuards(AuthGuard)
   @Query(/* istanbul ignore next */ () => Int)
-  async friendCount(@AuthenticatedUser() id: number) {
+  async friendCount(@Args('id', { type: /* istanbul ignore next */ () => Int }) id: number) {
     return (await this.userService.getFriends(id)).length;
   }
 
+  @UseGuards(AuthGuard)
   @Query(/* istanbul ignore next */ () => Int)
-  async hostingCount(@AuthenticatedUser() id: number) {
+  async hostingCount(@Args('id', { type: /* istanbul ignore next */ () => Int }) id: number) {
     return (await this.eventService.getEventsHostedByUser(id)).length;
   }
 
+  @UseGuards(AuthGuard)
   @Query(/* istanbul ignore next */ () => Int)
-  async portfolioCount(@AuthenticatedUser() id: number) {
+  async portfolioCount(@Args('id', { type: /* istanbul ignore next */ () => Int }) id: number) {
     return (await this.eventService.getUserEventsPortfolio(id)).length;
   }
 
   @Mutation(/* istanbul ignore next */ () => Boolean)
   async addFriend(
     @AuthenticatedUser() currentUserId: number,
-    @Args('userIdToAdd') userIdToAdd: number
+    @Args('userIdToAdd', { type: /* istanbul ignore next */ () => Int }) userIdToAdd: number
   ) {
     return this.userService.addFriend(currentUserId, userIdToAdd);
   }
 
   @Mutation(/* istanbul ignore next */ () => Boolean)
   async removeFriend(
     @AuthenticatedUser() currentUserId: number,
-    @Args('userIdToRemove') userIdToRemove: number
+    @Args('userIdToRemove', { type: /* istanbul ignore next */ () => Int }) userIdToRemove: number
   ) {
     return (
       this.userService.removeFriend(currentUserId, userIdToRemove) &&
       this.userService.removeFriend(userIdToRemove, currentUserId)
     );
   }
 
+  @Mutation(/* istanbul ignore next */ () => Boolean)
+  async removePendingRequest(
+    @AuthenticatedUser() currentUserId: number,
+    @Args('userIdToRemove', { type: /* istanbul ignore next */ () => Int }) userIdToRemove: number
+  ) {
+    return this.userService.removeFriend(currentUserId, userIdToRemove);
+  }
+
```

Comment:
💡I'd say this isn't needed, you can use the `removeFriend` mutation wherever you're using `removePendingRequest`, it'll behave in the same way

Reply:
I still preferred a separate mutation just for the semantic name